src/main/java/org/folio/rest/impl/PutOrderLineByIdHelper.java

```diff
@@ -199,19 +200,27 @@ public boolean isUpdateSuccessful() {
 		handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger)
       .thenAccept(body -> {
         PieceCollection pieces = body.mapTo(PieceCollection.class);
-
+        
         // Extract item Id's which already associated with piece records
         List<String> existingItemIds = pieces.getPieces()
             .stream()
             .map(Piece::getItemId)
             .filter(StringUtils::isNotEmpty)
             .collect(Collectors.toList());
-
+        
+        // Create pieces for successfully created item Ids
+        if(!itemIds.isEmpty()) {
```

Comment:
This check does not cover P/E Mix order format when some electronic resources are not expected to be registered in the inventory. The items will be created for physical resources only and the logic will not create piece records for electronic resources.
I propose to enhance the logic and calculate sum of `compPOL.getCost().getQuantityPhysical()` and `compPOL.getCost().getQuantityElectronic()` as total items and then subtract inventory quantity `HelperUtils#calculateInventoryItemsQuantity(pol)`
Ideally the logic should prepare POL content setting `compPOL.getCost().setQuantity(..)` in the beginning of the POL processing so this is always up to date...

Reply:
Thanks for your insight @piotr-kalashuk. Just curious, why should we `getQuantityPhysical()` and `getQuantityElectronic()` from `cost`?

Reply:
Please have a look at https://issues.folio.org/browse/UIOR-121. The Physical and/or Electronic quantities are specified in Cost details and locations' should be based on these figures.
I just checked and see that there is no `compPOL.getCost().setQuantity(..)` according to schema. So the sum should be used all the time to get overall quantity